### PR TITLE
Feature/answers to test handling

### PR DESCRIPTION
**Pull request by tsiananel**
CPT2108131350-1148x907
image
image

### CODE REVIEW

File: src/app/interfaces/essay-speaking.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
 export interface TopicModule {
-  id: number;
+  id: string;
   topicName: string;
 }
+
+// export interface Speaking
```
Comment (RomPavel) [inline src/app/interfaces/essay-speaking.ts:6]: remove this

File: src/app/models/writing-test/writing-test.component.ts [hunk 2]
```diff
@@ -29,4 +31,14 @@ export class WritingTestComponent implements OnInit {
     this.form.disable();
     this.disabled = true;
   }
+
+  clear() {
+    this.form.reset();
+  }
+
+  nextTest() {}
```
Comment (RomPavel) [inline src/app/models/writing-test/writing-test.component.ts:39]: remove this

File: src/app/models/writing-test/writing-test.component.ts [hunk 3]
```diff
@@ -29,4 +31,14 @@ export class WritingTestComponent implements OnInit {
     this.form.disable();
     this.disabled = true;
   }
+
+  clear() {
```
Comment (RomPavel) [inline src/app/models/writing-test/writing-test.component.ts:35]: do we need clear?

File: src/app/services/store/test-store.service.ts [hunk 4]
```diff
@@ -14,15 +14,15 @@ export class TestStoreService {
 
   allTests$ = this.allTestsSubject$.asObservable();
 
-  public resultsSubject$ = new BehaviorSubject<TestResult[] | null>(null);
+  public requestSubject$ = new BehaviorSubject<TestSubmit | null>(null);
 
-  testResults$ = this.resultsSubject$.asObservable();
+  requestBody$ = this.requestSubject$.asObservable();
```
Comment (RomPavel) [inline src/app/services/store/test-store.service.ts:19]: naming is too abstract

File: src/app/components/stepper/stepper.component.ts [hunk 5]
```diff
@@ -14,20 +16,47 @@ import { Question } from '../../interfaces/question-answer';
...
   ],
 })
-export class StepperComponent implements OnInit, OnChanges {
-  @Input() questionList!: Question[] | null | undefined;
+export class StepperComponent implements OnChanges, OnInit {
+  @Input() questionList: Question[] | null = null;
+
+  @Output() answersChosenId = new EventEmitter<string[] | null>();
+
+  listOfId: string[] | undefined = [];
```
Comment (RomPavel) [inline src/app/components/stepper/stepper.component.ts:24]: do you use it

Comment (tsiananel) [reply src/app/components/stepper/stepper.component.ts:24]: image
yes, in grammar & listening templates

File: src/app/components/stepper/stepper.component.ts [hunk 6]
```diff
@@ -14,20 +16,45 @@ import { Question } from '../../interfaces/question-answer';
...
         stepCtrl: ['', Validators.required],
       }),
     );
+    this.onChanges();
+  }
+
+  onChanges(): void {
+    const observables = this.stepperFormGroups?.map((form) =>
+      form.controls.stepCtrl.valueChanges.pipe(
+        startWith(''), // set initial value to let the subscribe to be called
```
Comment (ExxZzyy) [inline src/app/components/stepper/stepper.component.ts:44]: Remove comment

**Review by ExxZzyy — APPROVED** [review]

**Review by casterweb — APPROVED** [review]

**Pull request merged**